content/webapp/components/OnThisPageAnchors/index.tsx

```diff
@@ -223,76 +381,144 @@ const OnThisPageAnchors: FunctionComponent<Props> = ({
     );
     buttonRef.current.setAttribute(
       'aria-expanded',
-      isListActive ? 'false' : 'true'
+      isListActive ? 'true' : 'false'
     );
   }, [isListActive]);
 
+  useEffect(() => {
+    setActiveLinkText(
+      links.find(link => link.url.replace('#', '') === activeId)?.text ||
+        titleText
+    );
+  }, [activeId]);
+
   return (
-    <Root $isSticky={isSticky} $hasBackgroundBlend={hasBackgroundBlend}>
-      <h2
-        className={`${fontStyle} ${isSticky ? 'is-hidden-s is-hidden-m' : ''}`}
+    <>
+      <BackgroundOverlay
+        $isActive={isListActive}
+        onClick={() => setIsListActive(false)}
+      />
+      <div ref={onThisPageAnchorsStickyRef}></div>
+      <Root
+        $isSticky={isSticky}
+        $hasBackgroundBlend={hasBackgroundBlend}
+        $hasStuck={hasStuck}
       >
-        {titleText}
-      </h2>
-      {isSticky && (
-        <MobileNavButton
-          ref={buttonRef}
-          onClick={() => setIsListActive(!isListActive)}
+        <h2
+          className={`${fontStyle} ${isSticky ? 'is-hidden-s is-hidden-m' : ''}`}
```

Comment:
`fontStyle` is only ever used there (I think) and has a similar `isSticky` condition, could we merge both?